integration/shareddata/benchmark.go

```diff
@@ -48,3 +58,37 @@ func generateMixedValues() func() bson.D {
 
 	return gen
 }
+
+// generateLargeDocument returns generator that generates single deterministic document
+// with 200 elements that contain simple strings, doubles and objects.
+func generateLargeDocument() func() bson.D {
+	values := []any{
+		"foo", 42, "42", bson.D{{"42", "hello"}},
+	}
+	valuesLen := len(values)
+
+	elements := make([]bson.E, 200)
+	elements[0] = bson.E{"_id", 0}
+
+	for i := 1; i < len(elements); i++ {
+		elements[i] = bson.E{
+			Key:   fmt.Sprint(i),
+			Value: values[i%valuesLen],
+		}
+	}
+
+	doc := bson.D(elements)
+
+	var done bool
+	gen := func() bson.D {
+		if done {
+			return nil
+		}
+
+		done = true
+
+		return doc
+	}
+
```

Comment:
What is the purpose of that part of the code?

Reply:
To stop iterator (i.e. make it return `errIteratorDone`) the generator must to return `nil`. It may look weird on a single document generator, but it makes sense when generator returns 400 of documents.